sdk/eventhub/Azure.Messaging.EventHubs.Processor/tests/Diagnostics/DiagnosticsTests.cs

```diff
@@ -49,12 +48,18 @@ public async Task CheckpointManagerCreatesScope()
             var context = new MockPartitionContext("partition");
             var data = new MockEventData(new byte[0], sequenceNumber: 0, offset: 0);
 
-            var processor = new EventProcessorClient(Mock.Of<BlobContainerClient>(), "cg", endpoint.Host, eventHubName, fakeFactory, null);
+            var storageManager = new Mock<PartitionManager>();
+            var eventProcessor = new Mock<EventProcessorClient>(Mock.Of<BlobContainerClient>(), "cg", endpoint.Host, eventHubName, fakeFactory, null);
 
-            // TODO: find a way to call UpdateCheckpointAsync.
+            storageManager
+                .Setup(manager => manager.UpdateCheckpointAsync(It.IsAny<Checkpoint>()))
+                .Returns(Task.CompletedTask);
 
-            await Task.CompletedTask;
-            // await processor.UpdateCheckpointAsync(data, context);
+            eventProcessor
+                .Setup(processor => processor.CreateStorageManager(It.IsAny<BlobContainerClient>()))
+                .Returns(storageManager.Object);
+
+            await eventProcessor.Object.UpdateCheckpointAsync(data, context);
```

Comment:
Not critical for this PR, but the `PartitionPumpCreatesScopeForEventProcessing` that follows is something we should look at next week.   I think we can use the extension point that was added for the consumer client there and patch that up.